src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJLibrary.java

```diff
@@ -0,0 +1,58 @@
+package fi.aalto.cs.apluscourses.intellij.model;
+
+import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.roots.OrderRootType;
+import com.intellij.openapi.roots.impl.libraries.LibraryEx;
+import com.intellij.openapi.roots.libraries.LibraryProperties;
+import com.intellij.openapi.roots.libraries.LibraryTable;
+import com.intellij.openapi.roots.libraries.PersistentLibraryKind;
+import fi.aalto.cs.apluscourses.model.Library;
+import java.nio.file.Path;
+import org.jetbrains.annotations.CalledWithWriteLock;
+import org.jetbrains.annotations.NotNull;
+
+public abstract class IntelliJLibrary<
+    K extends PersistentLibraryKind<? extends LibraryProperties<S>>, S> extends Library {
+
+  @NotNull
+  protected final APlusProject project;
+
+  public IntelliJLibrary(@NotNull String name, @NotNull APlusProject project, int state) {
+    super(name, state);
+    this.project = project;
+  }
+
+  @CalledWithWriteLock
+  private void loadInternal() {
+    LibraryTable.ModifiableModel libraryTable = project.getLibraryTable().getModifiableModel();
+    com.intellij.openapi.roots.libraries.Library.ModifiableModel library = libraryTable
```

Comment:
not something to address now, but I suggest we consider codebase-wide change, where we somehow (by convention) distinct entities that are special to A+ from the same-named IntellijJ's ones (or framework ones, like Swing's):
`Project` (IJ) & `APlusProject`
`Module` (IJ) & `APlusModule`
`Component` (Swing) & `APlusComponent`
`Library` (IJ) & `APlusLibrary`
we could also take into account the fact, that we have **Model** entities, that are specific to IJ platform (like `IntelliJModule` etc.), so to have a uniform approach to this matter